### DISCUSSION TITLE
LOG4M UPLOAD TO CLOUDLOG???

### DISCUSSION THREAD

**Discussion by ChrisL79** [magicbug/Cloudlog#2802] [Q&A]
Hello. Is there anyway to upload qsos from log4om using the ADIF POST option? 

I'm guessing API key, but what would be the syntax for that?
What details do i need to input besides the server address?

If log4om will not work, what is the best and quickest way to push uploads other than cloudlog offline app which works flawlessly.

Thanks!

**Comment by magicbug** [comment]
You'd need to ask that support from Log4OM not something we support.

**Comment by ChrisL79** [comment]
Ok, so what is the best program to use to send logs other than direct input or cloudlogoffline.

Using windows here.

**Reply by HB9HIL** [reply]
I don't know Log4OM but probably the easiest way would be to export the QSO as ADIF from Log4om and import in in Cloudlog manually.

**Reply by bgritzmacher** [reply]
When I migrated from Log4OM to Cloudlog, I just did an ADIF export from Log4OM and imported it in Cloudlog.

**Comment by ChrisL79** [comment]
Thanks. Trying to avoid the manual part though.

**Comment by ChrisL79** [comment]
This is the way i hqve done it so far. I just wondered if there was an easier automatic way. How about cloudlogcat? I have tried Aurora, but no idea what i am doing with it or how to interface it to another program.

**Reply by HB9HIL** [reply]
## CloudlogCat
This Programm is able to read Frequency and Mode from your radio over a COM Port and pipe it to Cloudlog. With that you are able to log the real Frequency and Mode in QSO window. Also you have the possibility to pipe information from Sat32PC to Cloudlog.

## Aurora
This Programm handles mainly piping qso from WSJT-X to Cloudlog. This could be an option you could try IF Log4OM has the possibility to stream the QSO data over an UDP stream. But as mentioned. I don't know Log4OM and it's capabilities.

**Comment by ChrisL79** [comment]
Lovely, thank you. I have CloudlogCAT talking to the server, just need to interface with radio. Tomorrows project.

I will ask support at log4om if they would maybe add a Cloudlog upload feature.

The only thing Cloudlog needs is its own logging program, or to be included in the most popular ones.

**Reply by HB9HIL** [reply]
.. Ehm you can go to QSO -> Live QSO and log every QSO from here live and in color. What else is missing for you?

**Reply by ChrisL79** [reply]
Because it is a ballache. To do that i would have to fill out the qso tab, switch to the station tab and enter the rx and tx freq as well as input the rest of the fields. Why would you do that when it can be automated? Nobody ever swaps their 75 inch led tv for a 12" black and white portable. What is it i am missing? I am asking for help with automated logging, not manual. Thanks for your input in explaining CloudlogCat and Aurora. That was a good explanation.

**Reply by HB9HIL** [reply]
@ChrisL79 
I'm really sorry to say that but it seems like you are using Cloudlog not correctly. I can log a QSO with all necessary data in 5 Seconds....

Please write me an email, my mail is on QRZ.com. I would like to give you an overview about Cloudlog and it's capabilities.

**Comment by ChrisL79** [comment]
Thanks. I could do that, but if anyone has the same question in the future, they will not know how to resolve it if you help me via email. Not everyone arrives at Cloudlog with a lot of programming or Linux experience, so stupid questions will be asked.

99% of my operating is remote.

Cloudlogoffline takes care of that uploading ssb contacts immediately to my cloudlog server.

I was just womdering if there was a windows program for digi. Cloudlogcat seems to be the answer for digi stuff. So i will set that up tomorrow.

I do not mind logging manually with live qso. What i would like is all the important stuff on one tab. Switching tabs to input freqs twice does not seem to make any sense. Unless i am missing something very obvious, like configuring the panel to my needs. Is is what you need to do?

While we are talking about manual inputting, the qrz lookup does not work for me at all. 

I am aware for full details to be populated needs a sub to QRZ.

I fill out my qrz username and password in the config file, change callbook to qrz, use fullname true. 

Selecting callbook to qrz results in a timeout error once a call has been input. The only way to get Cloudlog back online again is to changeback to hamqth and a restart of apache.

**Reply by HB9HIL** [reply]
For logging Frequencies:

If you choose a band and a mode the default frequency will be automatically set as frequency. There is no need to fill the frequency to every kHz exact. If you want the exact frequency and don't want to put it in manually, you have to setup CloudlogCAT, connect your computer with your radio and you can automatically populate QRG and mode to your Cloudlog Instance. So you can manually log live QSO with exact QRG and Mode. If you don't do Cross-QRG or Crossband action there is no need to fill out RX Frequency. 

CloudlogCat does not push QSO data to your server. If you need a UDP stream for QSO data you should check out [Cloudlog Aurora](https://aurora.cloudlog.org/). 

For the QRZ Issue:

Can you please check the logs of your Cloudlog Instance? You can configure the Log Threshold in the config.php. Default path is in 'application/logs'. It's difficult to say what's the problem without further information. We don't know about a broken lookup at the moment. Also interesting to see would be on the debug page which version and migration you are on. 

vy 73 HB9HIL

**Comment by ChrisL79** [comment]
An update.
Used Aurora with JTDX. Sometimes it will upload, sometimes not. For every 5 qsos, say 3 or thereabouts will not upload. Any idea why?

Also the freq logged by log4om was 10.136 and when uploaded on my Cloudlog it reports a frequency of 10.138. Any ideas about this?

Also still struggling to upload my .adi log. Still having 504 timeout error. Apache time out is set at 9999. Any ideas?

I can give someone ssh login info if you want to check things.
